### PR TITLE
Move TPM2 integrations tests to use simulator

### PR DESCRIPTION
**Pull request by josephlr**
This PR now makes it so integration tests run against the simulator by default, instead of being skipped.
This required:
  
  - moving the integration tests to their own package (to prevent an import loop)
  - exposing `Encode` and `DecodeName` for the `Name` structure
  - moving some test cases between files  
  - a bunch of smaller, minor changes

EDIT: Given the nature of this PR, reviewing commit by commit should be the easiest. As each individual change is not that large.

@twitchy-jsonp @awly, do any changes need to be made to the CI to account for the fact that tests are now in `tpm2` and `tpm2/test`?

### CODE REVIEW

**Comment by twitchy-jsonp** [comment]
CI runs the following from the project root:

```shell
go build github.com/google/go-tpm/...
go test github.com/google/go-tpm/...
```

I think thats sufficient?

I have an open action item to finish implementing TPM 1.2 CI integration.

File: tpm2/structures.go [hunk 1]
```diff
@@ -888,7 +888,8 @@ type Name struct {
 	Digest *HashValue
 }
 
-func decodeName(in *bytes.Buffer) (*Name, error) {
+// DecodeName deserializes a Name hash from the TPM wire format.
```
Comment (twitchy-jsonp) [inline tpm2/structures.go]: Maybe mention this is a `TPM2B_NAME` structure explicitly?

**Comment by josephlr** [comment]
> CI runs the following from the project root:
> 
> ```shell
> go build github.com/google/go-tpm/...
> go test github.com/google/go-tpm/...
> ```
> 
> I think thats sufficient?

We will also need to run `go get -d -t -u github.com/google/go-tpm/...` to properly download any necessary dependencies (i.e. the simulator). That seems to be the reason the CI is failing right now.

File: tpm2/structures.go (hunk 1 above)
Comment (josephlr) [reply tpm2/structures.go]: Done,  added a comment on the struct itself.

**Pull request merged**